src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java

```diff
@@ -134,25 +137,17 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
      */
     private Issue createJiraIssue(AbstractBuild<?, ?> build, String filename) throws IOException, InterruptedException {
 
-        EnvVars environmentVariable = build.getEnvironment(TaskListener.NULL);
-
-        String buildURL = environmentVariable.get("BUILD_URL");
-        String buildNumber = environmentVariable.get("BUILD_NUMBER");
-        String jobName = environmentVariable.get("JOB_NAME");
-        String jenkinsURL = Jenkins.getInstance().getRootUrl();
+        EnvVars vars = build.getEnvironment(TaskListener.NULL);
 
+        String summary = String.format("Build %s failed", getBuildName(vars));
         String checkDescription = (this.testDescription.equals("")) ? "No description is provided" : this.testDescription;
-        String description = String.format("The test %s has failed. \n\n%s\n\n* First failed run : [%s|%s]\n** [console log|%s]",
-                jobName, checkDescription, buildNumber, buildURL, buildURL.concat("console"));
-
-        List<Component> components = getJiraComponents(build, this.component);
-
-        String summary = "Test " + jobName + " failure - " + jenkinsURL;
+        String description = String.format("%s\n\nThe build %s has failed.\nFirst failed run: %s",
+                checkDescription, getBuildName(vars), getBuildDetailsString(vars));
+        List<BasicComponent> components = getJiraComponents(build, this.component);
```

Comment:
there is some code in `getJiraComponents()` converting `List<Component>` to `List<BasicComponent>` and there is same code in `session.createIssue()` converting it back :) i think you could fix it